File: test_data/code/python/1446-consecutive-characters.py
class Solution:
    def maxPower(self, s: str) -> int:
        max_power = 0
        i = 0
        while i < len(s):
            count = 1
            while i + 1 < len(s) and s[i] == s[i + 1]:
                i += 1
                count += 1
            max_power = max(max_power, count)
            i += 1
        return max_power

File: test_data/code/java/1446-consecutive-characters.java
class Solution {
    public int maxPower(String s) {
        int maxPower = 0;
        char currentChar = s.charAt(0);
        int currentCount = 1;
        
        for (int i = 1; i < s.length(); i++) {
            if (s.charAt(i) == currentChar) {
                currentCount++;
            } else {
                maxPower = Math.max(maxPower, currentCount);
                currentChar = s.charAt(i);
                currentCount = 1;
            }
        }
        
        return Math.max(maxPower, currentCount);
    }
}